### PR TITLE
Added documentation for the portaal taak plugin

### PR DESCRIPTION
**Pull request by ivo-ritense**

### CODE REVIEW

**Review by theo-ritense — APPROVED** [review]

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+The portaal taak plugin is used to create taak objects in the Objecten API for a BPMN user task so they can be 
+completed through other applications like the NL Portal. This allows Valtimo to delegate tasks to other systems
+that are better suited to execute these tasks, or allow access from users that do not have access to the Valtimo
+frontend.
+
+## How does the plugin work
+
+The lifecycle of a portaal taak is as follows:
+1. When a process instance reaches a user task configured with the [Create Portaal taak](#create-portaal-taak) action
+an instance of the task is created in valtimo as usual. Additionally a taak object is created in the Objecten API of
```
Comment (ThomasRitense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:12]: Capital Valtimo

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+object type `Taak`. The status of the taak object will be `open`. The taak object includes information on who the 
+intended recipient is. This is the person or organization that is supposed to complete the task.
+2. External applications, such as the NL Portal, can retrieve the task information and offer a way to 
+complete the task. Often this will be a form to input data. 
+3. When the user completes the task in the external system, this system will update the taak object with the new data
+and change the status in the object to `ingediend`.
+4. The Objecten API should send a notification to the Notificaties API indicating the the object has been updated.
+5. Valtimo should be subscribed to notifications from the Notificaties API as part of the plugin setup and will receive
+the notification. When an update notification for a taak object is received Valtimo will check the status. If this is 
+set to `ingediend` the updated data from the taak object will be stored, and the handling process will be started.
```
Comment (ThomasRitense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:22]: Stored where?

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+action will complete the task in Valtimo and update the zaak object status to `verwerkt`, indicating the the taak has 
+been processed. 
+   
+## Configure the plugin
+
+A plugin configuration is required before the plugin can be used. A general description on how to configure
+plugins can be found [here](../configure-plugin.md).
+
+If the Portaal taak plugin is not visible in the plugin menu, it is possible the application is missing a dependency.
+Instructions on how to add the Zaken API dependency can be found
```
Comment (ThomasRitense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:33]: Zaken API?

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+[here](/getting-started/modules/zgw/portaaltaak.md).
+
+To configure this plugin the following properties have to be entered:
+- **Notification API plugin.** Reference to another plugin configuration that will be used to receive notification on
```
Comment (ThomasRitense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:37]: Shouldn't 'Notification API' be in Dutch?

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+happened, the actual completion of the task is also a plugin action that can be configured in this process.
+
+The process that is started in configured in the plugin properties by setting the 'Uploaded documents handler process'
+property. Valtimo ships with the `Process Portaaltaak uploaded Documents` process which has two user tasks.
+
+![Process Portaaltaak uploaded Documents](img/process-portaal-taak.png)
+
+These user tasks need to be configured with process links before the process can be used. The following actions should 
+be configured:
+- Link document to zaak - [Link Uploaded Documenten API document to Zaak](../zaken-api/configure-zaken-api-plugin.md#link-uploaded-document-to-zaak) in the Zaken API plugin
```
Comment (Klaas-Ritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:63]: I just tested it and the other link should be used here: `Zaken API - Link document to zaak`.

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 4 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:37]: Arguably. I used the variable names as they are displayed on the screen when using the english language settings.

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 1 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: Fixed

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 2 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: Fixed

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 3 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: Oops, fixed

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
+# Using the Portaal taak Plugin
+
+The Portaal Taak plugin is used to generate "taak" (task) objects within the Objecten API for a BPMN 
+user task. As a result, such tasks can be completed via external applications, such as the NL Portal. This feature
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:4]:
```suggestion
user task. These tasks can then be completed via external applications, such as the NL Portal. This feature 
```

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
+# Using the Portaal taak Plugin
+
+The Portaal Taak plugin is used to generate "taak" (task) objects within the Objecten API for a BPMN 
+user task. As a result, such tasks can be completed via external applications, such as the NL Portal. This feature 
+empowers Valtimo to assign tasks to other systems that are more adept at performing them. Furthermore, it enables users
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:5]:
```suggestion
empowers Valtimo to assign tasks to other systems that are more adept at performing them in a different context. Furthermore, it enables users 
```

I think the line as it is right now makes it seem like Valtimo isn't good for completing tasks. Not sure if that's the message we want to convey...

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 6 above)
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:4]: NL Portal isn't something that has been explained yet. I think giving examples is good, I just don't know if it makes sense to mention the NL Portal, since the intention was to not document portal related stuff (obviously stuff that's in Valtimo is the exception to this).

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+who lack access to the Valtimo frontend to access these tasks.
+
+## How does the plugin work
+
+The lifecycle of a portaal taak is as follows:
+1. When a process instance reaches a user task configured with the [Create Portaal taak](#create-portaal-taak) action
+an instance of the task is created in Valtimo as usual. Additionally a taak object is created in the Objecten API of
+object type `Taak`. The status of the taak object will be `open`. The taak object includes information on who the 
+intended recipient is. This is the person or organization that is supposed to complete the task.
+2. External applications, such as the NL Portal, can retrieve the task information and offer a way to
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:15]: NL Portal isn't something that has been explained yet. I think giving examples is good, I just don't know if it makes sense to mention the NL Portal, since the intention was to not document portal related stuff (obviously stuff that's in Valtimo is the exception to this).

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+definition to handle the completion itself. This can be done using the [Complete Portaal taak](#complete-portaal-taak) 
+plugin action. The process 'Process Portaaltaak uploaded Documents' that is shipped with Valtimo can be used here. See 
+[this section](#configuring-the-task-completion-process ) on how to set up this process. 
+
+An example plugin configuration:
+![example plugin configuration](img/configure-plugin.png)
+
+## Configuring the task completion process 
+
+When a task object is updated and Valtimo gets the notification, a process is started to handle additional steps needed
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:53]:
```suggestion
When a task object is updated and Valtimo receives the notification, a process is started to handle additional steps needed 
```

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+An example plugin configuration:
+![example plugin configuration](img/configure-plugin.png)
+
+## Configuring the task completion process 
+
+When a task object is updated and Valtimo gets the notification, a process is started to handle additional steps needed 
+for completing the task. Because the task can only be considered completed when all other processing of data has 
+happened, the actual completion of the task is also a plugin action that can be configured in this process.
+
+The process that is started in configured in the plugin properties by setting the 'Uploaded documents handler process'
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:57]: A bit confused as to what you're trying to say here.

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+property. Valtimo ships with the `Process Portaaltaak uploaded Documents` process which has two user tasks.
+
+![Process Portaaltaak uploaded Documents](img/process-portaal-taak.png)
+
+These user tasks need to be configured with process links before the process can be used. The following actions should 
+be configured:
+- Link document to zaak - [Link document to zaak](../zaken-api/configure-zaken-api-plugin.md#link-document-to-zaak) in the Zaken API plugin
+- Update Portaal Taak Status - [Complete Portaal taak](#complete-portaal-taak) in the Portaal Taak plugin
+
+Instead of using the `Process Portaaltaak uploaded Documents` process it is possible to create another process that will
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:67]: Maybe add a header here, e.g. `### Custom process`

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+multi-instance elements to iteratte over the list.
+
+![img.png](img/document-urls-collection-example.png)
+
+## Available actions
+
+The portaal taak plugin supports the following actions that can be configured in process links in order to create and 
+complete user tasks through external systems like the NL Portal.
+
+A general description on how to create process links, can be found [here](../create-process-link.md).
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:84]:
```suggestion
A general description on how to create process links can be found [here](../create-process-link.md).
```

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+### Create Portaal taak
+        
+The **Create portal task** action is linked to a user task that can be completed in an external system. A taak object
+is created when the task is reached. This taak object is created in the Objecten API configured for the object
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:89]: You're almost saying the same thing twice here. Maybe rephrase to something like:

`A taak object is created when the task is reached, in the Objecten API configured for the object management configuration for the plugin configuration.`

Not sure what the last half of this sentence should be by the way. I just copied what you stated, but I don't understand what you're trying to say.

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+A general description on how to create process links, can be found [here](../create-process-link.md).
+
+### Create Portaal taak
+        
+The **Create portal task** action is linked to a user task that can be completed in an external system. A taak object
+is created when the task is reached. This taak object is created in the Objecten API configured for the object 
+management configuration for the plugin configuration.
+
+When creating a process link the following properties have to be entered:
+- **Form type.** The type of form to be used for completing the user task. Options are 'Form definition' to use a form
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:93]: Maybe list the two options (form definition, url) as separate bullets that are more indented?

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+### Create Portaal taak
+        
+The **Create portal task** action is linked to a user task that can be completed in an external system. A taak object
+is created when the task is reached. This taak object is created in the Objecten API configured for the object 
+management configuration for the plugin configuration.
+
+When creating a process link the following properties have to be entered:
+- **Form type.** The type of form to be used for completing the user task. Options are 'Form definition' to use a form 
+definition that the receiving system is supposed to understand, or 'URL' when using a url to the form definition.
+- **Formulier ID.** The identifier of the form in the receiving system. Required when using form type 'Form definition'.
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:95]: For this and the bullet below it, it might be worth mentioning they will only be shown when the corresponding form type has been selected.

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+object type `Taak`. The status of the taak object will be `open`. The taak object includes information on who the 
+intended recipient is. This is the person or organization that is supposed to complete the task.
+2. External applications, such as the NL Portal, can retrieve the task information and offer a way to 
+complete the task. Often this will be a form to input data. 
+3. When the user completes the task in the external system, this system will update the taak object with the new data
+and change the status in the object to `ingediend`.
+4. The Objecten API should send a notification to the Notificaties API indicating the the object has been updated.
+5. Valtimo should be subscribed to notifications from the Notificaties API as part of the plugin setup and will receive
+the notification. When an update notification for a taak object is received Valtimo will check the status. If this is 
+set to `ingediend` the updated data from the taak object will be stored based on the value resolver configuration, and
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:22]: Maybe link to the relevant section on how this can be configured?

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+of the user.
+
+An example process link configuration:
+![Create portal task process link](img/configure-create-portaal-taak.png)
+
+### Complete Portaal taak
+        
+The **Complete Portaaltaak** action should be used in the `Uploaded documents handler process` to do the last step in 
+handling the update from the objects API. When executed it will complete the user task and update the status of the 
+taak object to `verwerkt`
```
Comment (marijnritense) [inline using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:118]: Missing a dot at the end here. Is this the end of the documentation, or is there still some uncommited code? If there's nothing to configure, it might be worth mentioning there are no additional properties that have to be configured.

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 6 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: It's a tough one. I see your point, but i do think that it's the only actual example that we have right now. It makes the goal of this feature a bit more clear IMO. @ThomasRitense what do you think?

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 7 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: I do agree with the issue you have with it, but not the suggestion. I don't think it's the context that is different. I change the text, please check if this is ok.

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 14 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: Good idea

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 15 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: Fixed

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 16 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md:22]: I tried to find this but i don't think we have any documentation on this.

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 17 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: Fixed

File: using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md (hunk 10 above)
Comment (ivo-ritense) [reply using-valtimo/plugin/portaaltaak/configure-portaaltaak-plugin.md]: The grammar wasn't really correct. Please check again.

**Review by marijnritense — APPROVED** [review]
